src/charm.py

```diff
@@ -125,7 +127,13 @@ def _process_env_file(self, event):
         try:
             resource_path = self.model.resources.fetch("env-file")
             env = dotenv_values(resource_path)
-            self._state.env = env
+
+            for key, _ in env.items():
+                if key.startswith("TWC_"):
```

Comment:
i guess you could prefix these ones too, so they don't collide

Reply:
Environment variables provided by the user shouldn't be modified as the user will be expecting to fetch them by their original name within their code.